### DISCUSSION TITLE
Smooth scrolling for Sideband

### DISCUSSION THREAD

**Discussion by hardBSDk** [markqvist/Reticulum#134] [Requests for New Features]
Actual scrolling have stuttering on PC.

**Comment by markqvist** [comment]
Yes, there is a bit of performance issues with KivyMD currently. I think *some* of it comes from the fact that there is currently a GLSL shader bug that causes the app to crash on Adreno GPU, and hardware accelerated shaders are therefore completely disabled right now, until the bug is fixed.

**Reply by markqvist** [reply]
Other parts of it is simply because the widget structure is not very optimised, and I learned too late that KivyMD gets really slow when creating many widgets, and special care has to be taken to only keep a small amount of widgets active at the same time.

This can be improved by rewriting the list widgets to use an alternative class in Kivy that is more efficient.

**Reply by hardBSDk** [reply]
@markqvist I forgot to mention Python with GTK.

https://pygobject.readthedocs.io/en/latest/

Anaconda is a nice example, the Fedora installer.

https://fedoraproject.org/wiki/Anaconda

**Reply by markqvist** [reply]
Nice, thanks!
